worker/worker.go

```diff
@@ -649,14 +624,26 @@ func (w *worker) rhpFormHandler(jc jape.Context) {
 
 		contract, txnSet, err = RPCFormContract(ctx, t, renterKey, renterTxnSet)
 		if err != nil {
-			w.bus.WalletDiscard(ctx, renterTxnSet[len(renterTxnSet)-1])
+			ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancel()
+			discardErr := w.bus.WalletDiscard(ctx, renterTxnSet[len(renterTxnSet)-1])
+			if discardErr != nil {
+				w.logger.Errorf("failed to discard txn after failed formation: %v", discardErr)
+			}
 			return err
 		}
 		return
 	})
 	if jc.Check("couldn't form contract", err) != nil {
 		return
 	}
+
+	// broadcast the transaction set
+	err = w.bus.BroadcastTransaction(jc.Request.Context(), txnSet)
+	if err != nil && !strings.Contains(err.Error(), modules.ErrDuplicateTransactionSet.Error()) {
```

Comment:
Do we want to log the event of a `modules.ErrDuplicateTransactionSet`? Seems like it's going by unnoticed now but if we keep running into that we probably want to know about it right.

Reply:
We could debug log it if you want but it's not something out of the ordinary. It only means that the host was faster at broadcasting it.